bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaValues.java

```diff
@@ -85,8 +87,41 @@ public static ObjectValue createObjectValue(BPackage packageId, String objectTyp
             fields[j++] = fieldValues[i];
             fields[j++] = true;
         }
-        //passing scheduler, strand and properties as null for the moment, but better to expose them via this method
-        return valueCreator.createObjectValue(objectTypeName, null, null, null, fields);
+        return valueCreator.createObjectValue(objectTypeName, getScheduler(), getStrand(), null, fields);
```

Comment:
please remove `getScheduler` method, since `Scheduler.getStrand()` is costly (because it access a thread local) it should be called ones. assign the strand to a local var and access the scheduler form there.

Reply:
I modified considering all suggestions in [this](https://github.com/ballerina-platform/ballerina-lang/pull/22913/commits/515f7fe10aadde5656b821ccb180e31abba89504)